### DISCUSSION TITLE
Bug? Just got an email warning me about abuse-detection systems.

### DISCUSSION THREAD

**Discussion by AceVentura** [community/community#130825] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

> On behalf of the GitHub Security team, I want to first extend our gratitude for your continued use of GitHub and for being a valued member of the >GitHub community.
>
>Recent activity on your account caught the attention of our abuse-detection systems. This activity included use of Copilot that was indicative of >scripted interactions or of an otherwise deliberately unusual or strenuous nature. While we have not yet restricted Copilot access for your account, >further anomalous activity could result in a temporary suspension of your Copilot access.
>
>While I’m unable to share specifics on rate limits, we prohibit all use of our servers for any form of excessive automated bulk activity, as well as any >activity that places undue burden on our servers through automated means. Please refer to our Acceptable Use Policies on this topic: >https://docs.github.com/site-policy/acceptable-use-policies/github-acceptable-use-policies#4-spam-and-inauthentic-activity-on-github.
>
>Please also refer to our Terms for Additional Products and Features for GitHub Copilot for specific terms: https://docs.github.com/site-policy/github->terms/github-terms-for-additional-products-and-features#github-copilot.
>
>Sincerely,
>GitHub Security

I have no idea why I'm receiving this. I'm barely using copilot (mostly to generate ai messages for commits). 
I don't have any other sessions, and I have checked the security logs, nothing wrong with them, so it can't be someone else using my account.

This sounds like a bug, anyone else is getting this?

**Comment by cristi-envato** [comment]
Same for me, I received the message about an hour ago.

**Reply by x-typo** [reply]
Same.pretty sure I didn't spam the requests to trigger this kind of behavior...

**Comment by dse82** [comment]
Same here, also just two hours ago. I even received the same email again twenty minutes later. Also not using copilot much (at least actively to my knowledge. don’t know what vscodes integration might do by itself …).

Must be a bug in the abuse detection or in microsoft’s very own copilot-integration plugin.

**Comment by marty0678** [comment]
Just got the same thing and thought it was spam. I'm contacting support to verify as the email is a bit sus to confirm if this is legit. Can't imagine how I did this, but I wonder if it's a bug in VsCode somewhere.

I remember at one point I was rate limited from GitHub for a while because their Pull Request VsCode extension was spamming their API, triggering rate limits, and locked me out of my account. That went on for a week until they fixed it, so wonder if it's a similar thing here that we aren't aware of.

**Comment by jeremydavidson** [comment]
Same... using it during normal coding day, but nothing scripted, bulk, or excessive at least not intentionally. I wonder if we'll soon get a follow up like "sorry, this email went out to more users than intended".

**Comment by vstyler96** [comment]
Same, I received this email an hour ago! It was a regular day usage, or even less

**Comment by sgoedecke** [comment, marked as answer]
Hi folks, I work on abuse for GitHub. I can confirm that this was an error on our end, as many of you anticipated: we sent out warnings to a group of users that shouldn't have been warned. We've fixed the core problem - no more warnings are going out - and we've scrubbed the warnings from our system. Sorry about that!

**Reply by DavidZidar** [reply]
Hi, please email affected users about what happened and confirm that they did nothing wrong. I only found this thread because I was about to file a support request about this issue.

**Reply by marty0678** [reply]
Support responded to me saying it was a legit warning and they can't comment on specifics but seemed to imply I was doing something I shouldn't. I asked for clarity and linked this response. I am assuming support just hasn't been updated on this issue but will update when I hear back.

**Reply by anhduc130** [reply]
+1 to @DavidZidar , please send out another email to that "shouldn't have been warned group of users", confirming that they're not the target. Thanks!

**Reply by ellisio** [reply]
I have been noticing `Github > Copilot > Editor: Enable Code Actions`, if enabled, has been causing some extreme delay on save over the last couple days. If I turn this off, the delay stops and files save normally.

At first, I thought it was ESLint's extension slowing down (happens from time to time). But, receiving this email made me go digging because I don't use Copilot outside of day-to-day activities working on code.

Perhaps this is related to the "abuse"? Maybe a bug in the extension causing several API calls/retries?

**Reply by vstyler96** [reply]
Just to inform other users, I received an email it was an error :D

**Reply by david-priest** [reply]
I just got the same warning email. I think it may be related to the chattr plugin in R studio (https://github.com/orgs/community/discussions/153621).

**Reply by agnesevardanega** [reply]
Just received two email, opened a ticket, and the answer is 

> Hi Agnese,

> Thank you for contacting GitHub Support!

> Please verify you have received the email from GitHub Security warning you about your use of Copilot which goes against our Terms of Service. Please verify you have read and will adhere to the contents of the email.

> Please be aware that any subsequent violations of our Terms of Service and our GitHub Acceptable Use Policies will lead to the permanent revocation of your Copilot access and a permanent suspension of your GitHub account.

(Using RStudio and VS Code, nothing special: I  teach basic data analysis and during the semester, probably, I am more active)

thank you

**Comment by benoxoft** [comment]
I got a similar email from github yesterday. I would like to respond to it.

I am using the helix editor with copilot. https://helix-editor.com/
I am using this LSP with helix to use copilot: https://github.com/leona/helix-gpt
100% of my activity is authentic.
I have disabled this LSP from my editor for the time being.

Is the use of this project in violation of point 6?
_You will not reproduce, duplicate, copy, sell, resell or exploit any portion of the Service, use of the Service, or access to the Service without our express written permission._
 
There is an open ticket in this project talking about triggering the abuse system: https://github.com/leona/helix-gpt/issues/63

I write this message in the hope that volunteers can implement their own client for copilot and that users like me can use them in place of the other commercial software. Is there something the author of the project can do so that we can use their LSP in Helix?

Thank you very much.

**Comment by pxhdev** [comment]
I also got two warning email. I was using zbirenbaum/copilot.lua in neovim, and coded in this environment for over 3 months and suddenly I got warnings.

**Reply by dexterklui** [reply]
I also just got the same warning using zibrenbaum/copilot.lua with neovim, and have been coded in this environment for about a year without any warning.

But I remember recently some errors with copilot did pop up a few times, mentioning something like reject query because hitting the limit.

I wonder if more people are facing the same issue.

**Reply by dexterklui** [reply]
I think this [issue#117 from zibrenbaum/copilot-cmp](https://github.com/zbirenbaum/copilot-cmp/issues/117#issue-2495359282) is related to the problem

**Comment by brianhuster** [comment]
I also got an email today (Oct 13 Indochina time). I am using 3 Neovim plugins related to Copilot : [copilot.vim](https://github.com/github/copilot.vim), [CopilotChat.nvim](https://github.com/CopilotC-Nvim/CopilotChat.nvim) and [avante.nvim](https://github.com/yetone/avante.nvim). Not sure which one causes the problem.

**Comment by eyoong43** [comment]
Hi All i am also having this issue all i used is vscode copilot and obsidian plugin copilot... Can any one help? Just got the trial...

**Reply by Elanngo-Madheswaran** [reply]
I am too affected by same

**Comment by pruthvi-hash** [comment]
I got this Today at 11 am IST. Now I am not able to use copilot. Any fixes?

**Comment by zzqxztc** [comment]
I also have this problem

**Comment by 2594806402** [comment]
May I ask how it can be done? Urgent

**Comment by dharitri-dessai** [comment]
I got this message for github saying .....  "On behalf of the GitHub Security team, I want to first extend our gratitude for your continued use of GitHub and for being a valued member of the GitHub community.

Recent activity on your account has caught the attention of our abuse-detection systems. This activity may have included use of Copilot via scripted interactions, an otherwise deliberately unusual or strenuous nature, or use of unsupported clients or multiple accounts to circumvent billing and usage limits.

Due to this, we have suspended your access to Copilot. "

But I don't understand what that exactly means. Can you please elaborate on this, is the payment failing? Can you check my copilot subscription? Is there anything wrong in there? 

Let me know how to activate the copilot subscription? I was using a copilot for a long time now . And suddenly this message comes.

I will appreciate any help in this regards.

**Comment by KaDeEm21** [comment]
I have that kind of email too. How can I appeal to it or what can I do?

**Reply by lauwah** [reply]
Me too and sent an email to ask for re-enabling the account.
